pxtlib/emitter/ir.ts

```diff
@@ -310,6 +312,38 @@ namespace ts.pxt.ir {
                 }
             }
         }
+
+        get isTemporary(): boolean {
+            return false
+        }
+    }
+
+    //Cells that represent variables that are generated by the compiler as temporaries
+    //The user cannot access these cells from JS or blocks
+    export class UnnamedCell extends Cell {
+        private static unnamedCellCounter = 0
+        private uid: number
+
+        constructor(public index: number, public owningProc: Procedure) {
+            super(index, null, null)
+            this.uid = UnnamedCell.unnamedCellCounter++
+        }
+
+        getName() {
+            return "unnamed" + this.uid
+        }
+
+        uniqueName() {
+            return this.getName() + "___" + this.index
+        }
+
+        isByRefLocal() {
+            return this.isLocal()
```

Comment:
By-ref local means it is captured by an inner function and written subsequently, which means it has to be compiled as a one-field record (feel free to add some comments to that effect). These are never like that, so just return 'false'.